src/inet/UDPEndPointImplSockets.h

```diff
@@ -63,13 +63,12 @@ class UDPEndPointImplSockets : public UDPEndPoint, public EndPointStateSockets
 
 #if CHIP_SYSTEM_CONFIG_USE_PLATFORM_MULTICAST_API
 public:
-    using MulticastGroupHandler = CHIP_ERROR (*)(InterfaceId, const IPAddress &);
-    static void SetJoinMulticastGroupHandler(MulticastGroupHandler handler) { sJoinMulticastGroupHandler = handler; }
-    static void SetLeaveMulticastGroupHandler(MulticastGroupHandler handler) { sLeaveMulticastGroupHandler = handler; }
+    using MulticastGroupHandler = CHIP_ERROR (*)(InterfaceId, const IPAddress &, bool /* join */);
```

Comment:
could we use a named enum instead of a bool?

```cpp
enum class MulticastOperation {
   kJoin,
   kLeave,
}

using MulticastGroupHandler = CHIP_ERROR (*)(InterfaceId, const IPAddress &, MulticastOperation);
```

that way I can read `handler(interface_id, ip, MulticastOperation::kJoin)` which seems clearer than `handler(interface_id, ip, true)`